### PR TITLE
[Hs-183] My profile screen

### PR DESCRIPTION
**Pull request by DongNguyenQ**

### CODE REVIEW

File: lib/features/profile/my_profile/view_model/my_profile_state.dart [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  const MyProfileInitialState();
+}
+
+class GettingUserInformationState extends MyProfileState {
+  const GettingUserInformationState();
+}
+
+class GetUserInformationSucceedState extends MyProfileState {
+  final UserDetail user;
+  const GetUserInformationSucceedState(this.user);
```
Comment (thenartus) [inline lib/features/profile/my_profile/view_model/my_profile_state.dart:20]: this needs a `props` override

File: lib/features/profile/my_profile/view/my_profile_screen.dart [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    return Builder(builder: (context) {
+      return Scaffold(
+        appBar: AppBar(
+          title: Text(HatSpaceStrings.current.myProfile,
+              style: Theme.of(context)
+                  .textTheme
+                  .bodyMedium
+                  ?.copyWith(fontWeight: FontStyleGuide.fwBold)),
+          actions: [
+            IconButton(
```
Comment (thenartus) [inline lib/features/profile/my_profile/view/my_profile_screen.dart:37]: Why do u use `IconButton` for a Text button?

File: lib/features/profile/my_profile/view/my_profile_screen.dart [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+          ],
+          backgroundColor: HSColor.neutral1,
+          automaticallyImplyLeading: false,
+          leading: IconButton(
+            onPressed: () => context.pop(),
+            icon: SvgPicture.asset(Assets.icons.arrowCalendarLeft),
+          ),
+          elevation: 0,
+          bottom: PreferredSize(
+              preferredSize: const Size.fromHeight(1.0),
```
Comment (thenartus) [inline lib/features/profile/my_profile/view/my_profile_screen.dart:55]: `HsDimens.size1`

File: lib/features/profile/my_profile/view/my_profile_screen.dart [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+              child: Container(
+                color: HSColor.neutral2,
+                height: 1.0,
```
Comment (thenartus) [inline lib/features/profile/my_profile/view/my_profile_screen.dart:58]: `HsDimens.size1`

File: lib/features/profile/my_profile/view/my_profile_screen.dart [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+                  return '';
+                }, builder: (context, displayName) {
+                  return _InformationView(
+                    title: HatSpaceStrings.current.displayName,
+                    value: displayName,
+                  );
+                }),
+                BlocSelector<MyProfileCubit, MyProfileState, String>(
+                    selector: (state) {
+                  return '';
```
Comment (thenartus) [inline lib/features/profile/my_profile/view/my_profile_screen.dart:118]: add `TODO` here

File: lib/features/profile/my_profile/view/my_profile_screen.dart [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+                  .textTheme
+                  .bodyMedium
+                  ?.copyWith(fontWeight: FontWeight.w500)),
+        ] else ...[
+          Text(HatSpaceStrings.current.notUpdated,
+              style: Theme.of(context).textTheme.bodyMedium?.copyWith(
+                  fontWeight: FontWeight.w500, color: HSColor.neutral5)),
+        ],
+        const SizedBox(height: HsDimens.spacing12),
+        const Divider(color: HSColor.neutral2, thickness: 1, height: 1)
```
Comment (thenartus) [inline lib/features/profile/my_profile/view/my_profile_screen.dart:197]: `HsDimens.size1`

**Review by thenartus — COMMENTED** [review]
a few comments

File: lib/features/profile/my_profile/view/my_profile_screen.dart [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+    return Builder(builder: (context) {
+      return Scaffold(
+        appBar: AppBar(
+          title: Text(HatSpaceStrings.current.myProfile,
+              style: Theme.of(context)
+                  .textTheme
+                  .bodyMedium
+                  ?.copyWith(fontWeight: FontStyleGuide.fwBold)),
+          actions: [
+            TextButton(
```
Comment (suesitran) [inline lib/features/profile/my_profile/view/my_profile_screen.dart:37]: please use `TextOnlyButton`

**Review by suesitran — APPROVED** [review]

**Pull request merged**